### PR TITLE
Extra specs from mutant

### PR DESCRIPTION
**Pull request by kbrock**
Mutant showed a number of use cases that were not present. This does its best at filling in those gaps.

These were pulled out of #20 to make obvious that these are only here to add coverage, not test new behavior.

### CODE REVIEW

File: spec/awesome_spawn_spec.rb [hunk 1]
```diff
@@ -7,10 +7,10 @@
   shared_examples_for "run" do
     context "options" do
       it ":params won't be modified" do
-        params      = {:user => "bob"}
+        params      = {:params => {:user => "bob"}}
         orig_params = params.dup
-        allow(subject).to receive(:launch).and_return(["", "", 0])
-        subject.send(run_method, "true", :params => params)
+        allow(subject).to receive(:launch).with("true --user bob", {}).and_return(["", "", 0])
+        subject.send(run_method, "true", params)
```
Comment (Fryguy) [inline spec/awesome_spawn_spec.rb]: This actually changes the original spec.  Before it would test the inner hash wasn't modified, now it no longer tests that.

**Comment by Fryguy** [comment]
The rest looks good!  Sorry I took so long to review it.

**Comment by kbrock** [comment]
@Fryguy Good catch. I added a separate test to check that.
I think mutant was complaining about an edge case that we weren't testing there. So that is why I changed it. But yes, it looks like we were not testing the params changed - and that is more probable
